src/framework/filterControl/FilterItem.jsx

```diff
@@ -0,0 +1,122 @@
+
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+import {
+  Filter,
+} from '../services';
+
+import Entity from '../services/string/Entity';
+import FilterForm from './forms/FilterForm.jsx';
+
+export default class FilterItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      filter: this.props.filter,
+      isFormOpened: false,
+    };
+    this.onCancelEditing = this.onCancelEditing.bind(this);
+    this.onEditFilter = this.onEditFilter.bind(this);
+    this.onUpdateFilter = this.onUpdateFilter.bind(this);
+    this.onRemoveSelectedFilter = this.onRemoveSelectedFilter.bind(this);
+  }
+
+  onCancelEditing() {
+    this.setState({
+      isFormOpened: false,
+    });
+  }
+
+  onEditFilter() {
+    this.setState({
+      isFormOpened: true,
+    });
+  }
+
+  onRemoveSelectedFilter() {
+    this.props.onRemoveSelectedFilter(this.state.filter);
+  }
+
+  onUpdateFilter(oldFilter, filter) {
+    this.props.onReplaceFilter(oldFilter, filter);
+    this.setState({
+      filter,
+      isFormOpened: false,
+    });
+  }
+
+  renderFilterForm() {
+    const onUpdateFilter = filter =>
+      this.onUpdateFilter(this.state.filter, filter);
+    return (
+      <FilterForm
+        filterOption={this.state.filter.filterOption}
+        comparisonValue={this.state.filter.comparisonValue}
+        onAddFilter={onUpdateFilter}
+      />
+    );
+  }
+
+  renderForm() {
+    return (
+      <div className="filterFormDropdown filterFormDropdown--editing">
```

Comment:
I tried to combine editing and creating inside `FilterItem`, but it got too complicated. `CreateFilterFormDropdown` requires `FilterOption`, `FilterItem` requires `Filter` instead. Rendering would have to be different based on whether `FilterOption` or `Filter` was initially passed. In the end I decided  to separate two scenarios to make it cleaner and more straightforward.